src/main/java/de/unijena/cheminf/mortar/controller/MainViewController.java

```diff
@@ -473,30 +473,34 @@ private void closeWindowEvent(WindowEvent anEvent) {
      * @param aParentStage Stage where to open the file chooser dialog
      */
     private void chooseAndImportMoleculeFile(Stage aParentStage) {
-        if (!this.moleculeDataModelList.isEmpty()) {
-            if (!this.isFragmentationStopAndDataLossConfirmed()) {
-                return;
-            }
-            this.fragmentationService.clearCache();
-        }
         Importer tmpImporter = new Importer(this.settingsContainer);
         File tmpFile = tmpImporter.openFile(aParentStage);
-        this.importMoleculeFile(tmpFile);
+        this.importMoleculeFile(tmpFile, tmpImporter);
     }
     //
     /**
      * Loads molecule file and opens molecules tab.
+     * Convenient method to avoid using a null parameter for the importer.
      *
      * @param aFile File that contains molecular data
      */
     private void importMoleculeFile(File aFile) {
+        this.importMoleculeFile(aFile, new Importer(this.settingsContainer));
+    }
+    //
+    /**
+     * Loads molecule file and opens molecules tab.
+     *
+     * @param aFile File that contains molecular data
+     */
+    private void importMoleculeFile(File aFile, Importer anImporter) {
         if (!this.moleculeDataModelList.isEmpty()) {
             if (!this.isFragmentationStopAndDataLossConfirmed()) {
                 return;
             }
             this.fragmentationService.clearCache();
         }
-        Importer tmpImporter = new Importer(this.settingsContainer);
+        Importer tmpImporter = Objects.requireNonNullElseGet(anImporter, () -> new Importer(this.settingsContainer));
         if (Objects.isNull(aFile)) {
             return;
         }
```

Comment:
Please move this check up to be the first thing the method does. Alternatively, please add a return statement to the end of chooseAndImportMoleculeFile() if the "opened" file is null, i.e. the file chooser was cancelled. Or maybe do both to be safe.
Because right now, the confirmation dialog also appears if I click "cancel" in the file chooser. That is confusing.

Reply:
This check has nothing to do with the dialog. But I'll take care of it.

Reply:
I know, thanks, but it will keep the dialog from being opened when it shouldn't because of the "return" :D

Reply:
Thank you <3

Reply:
It's funny that we hadn't noticed this before. I never changed the order of the checks.

Reply:
Before, the confirmation dialog appeared before the file chooser was opened.